CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -246,4 +246,33 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
     return Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId];
   }
 
+  /**
+   * check if financial account is referenced by financial item
+   *
+   * @param $financialAccountId Integer
+   *
+  * @param $financialAccountTypeID Integer
+  *
+   * @return bool
+  *
+  */
+ public static function validateFinancialAccount($financialAccountId, $financialAccountTypeID = NULL) {
+   if (!$financialAccountId) {
```

Comment:
Sorry I think I was referring to !$financialAccountId - I just wonder if we really need to have that check. Seems unnecessary. It would only happen if there were bad code (i.e in dev not prod) & would fail on the query.